src/plugins/wizard/public/application/components/data_source_select.tsx

```diff
@@ -36,6 +37,7 @@ export const DataSourceSelect = () => {
         const foundOption = indexPatterns.filter((s) => s.id === option.id)[0];
         if (foundOption !== undefined && typeof foundOption.id === 'string') {
           dispatch(setIndexPattern(foundOption.id));
+          dispatch(setHasChange({ hasChange: true }));
```

Comment:
I'm not too sure that this approach to hasChange is scalable. We can easily miss certain edits if we do not update the state correctly. 

One way to do this right could be to subscribe to the store and update the state if the store has changed. See https://redux.js.org/api/store#subscribelistener